src/utils/GlobalStateProvider.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useReducer } from 'react';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
+type Action =
+  | { type: 'setUsers'; payload: User[] }
+  | { type: 'setComments'; payload: Comment[] }
+  | { type: 'setSelectedPost'; payload: Post | null }
+  | { type: 'setIsPostSelected'; payload: boolean }
+  | { type: 'setIsFormEnabled'; payload: boolean }
+  | { type: 'setSelectedUser'; payload: User | null };
+
+interface State {
+  users: User[] | null;
+  comments: Comment[];
+  selectedUser: User | null;
+  selectedPost: Post | null;
+  isPostSelected: boolean;
+  isFormEnabled: boolean;
+}
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'setUsers':
+      return {
+        ...state,
+        users: action.payload,
+      };
+
+    case 'setComments':
+      return {
+        ...state,
+        comments: action.payload,
+      };
+
+    case 'setSelectedPost':
+      return {
+        ...state,
+        selectedPost: action.payload,
+      };
+
+    case 'setSelectedUser':
+      return {
+        ...state,
+        selectedUser: action.payload,
+      };
+
+    case 'setIsPostSelected':
+      return {
+        ...state,
+        isPostSelected: action.payload,
+      };
+
+    case 'setIsFormEnabled':
+      return {
+        ...state,
+        isFormEnabled: action.payload,
+      };
+
+    default:
+      return state;
+  }
+}
+
+const initialState: State = {
+  users: null,
```

Comment:
```suggestion
  users: [],
```
set the empty array as default value